src/ReflectionParameter.php

```diff
@@ -329,7 +336,9 @@ public function isCallable()
      */
     public function isDefaultValueAvailable()
     {
-        return isset($this->parameterNode->default);
+        return
+            isset($this->parameterNode->default) &&
```

Comment:
Does this belong to this PR or should a separate bugfix PR with connected unit tests?

Reply:
Again, this is part of the handling for optional parameters without default values, which is new in this PR.

Reply:
Better to move this into separate PR for optional parameter without default value handling (part 5).

Reply:
OK. See above.